robot/server/server.go

```diff
@@ -67,10 +71,25 @@ func (s *Server) Tunnel(srv pb.RobotService_TunnelServer) error {
 		return fmt.Errorf("failed to receive first message from stream: %w", err)
 	}
 
+	var destAllowed bool
+	dialTimeout := defaultTunnelConnectionTimeout
+	// Ensure destination port is available; otherwise error.
+	for _, tte := range s.robot.TrafficTunnelEndpoints() {
+		if int(req.DestinationPort) == tte.Port {
+			destAllowed = true
+			if tte.ConnectionTimeout != 0 {
+				dialTimeout = tte.ConnectionTimeout
```

Comment:
Note that I have no error checking for specifying the same port multiple times in `traffic_tunnel_endpoints`. If a user does that, the fist `connection_timeout` in the proto will be honored.